### PR TITLE
show file anomalies on exposure group page

### LINKED ISSUE #1557: Indicate an anomaly is marked in some detectors
_Issue [JWQL-193](https://jira.stsci.edu/browse/JWQL-193) was created on JIRA by [Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert):_

Currently, when flagging an anomaly at the detector level (i.e. flag a Short in NRS1), the checkbox for that anomaly is checked on the detector page. However, when you look at the group page for that exposure (where you see all the detectors), the anomaly box is not checked. This is because the anomaly box on the group page is not checked unless all detectors in that group have that anomaly flagged.

However, this means that when browsing through data, if you are browsing at the group level, you will miss any flagged anomalies. 

Is there a way to indicate on the group page that a particular anomaly is present in one/some of the detectors in that group? Perhaps some third option besides having the box be unchecked or checked? e.g. if there is a dot in the center of the checkbox, then the anomaly is present in some of the detectors?

### PR DESCRIPTION
**Pull request by BradleySappington**
When viewing groups of exposures, show table of anomalies for each file.
Update how tables get populated to allow for dynamic setting of column pairs.

### CODE REVIEW

**Comment by BradleySappington** [comment]
Screenshot 2024-05-07 at 10 56 16 AM

**Comment by BradleySappington** [comment]
@bhilbert4 , I updated your table maker to designate the table id (so multiple tables per page can use it), and to take a user defined "nr_paired_columns" parameter.  So we can use that for any shape table that has key/value combo moving forward

**Comment by bhilbert4** [comment]
Nice! Looks good. Have you tested on a NIRCam group that has 10 detectors?

**Comment by mfixstsci** [comment]
@BradleySappington this looks great! I am curious along with @bhilbert4 about how that would look. @bhilbert4 do you know of an example off of the top of your head that will have a large number of detectors that Brad could test on?

**Comment by bhilbert4** [comment]
Looks like jw01022099001_03101_00001 should have all 10 nircam detectors, when running locally.

**Comment by mfixstsci** [comment]
> Looks like jw01022099001_03101_00001 should have all 10 nircam detectors, when running locally.

Thank you 🙏

**Comment by BradleySappington** [comment]
@mfixstsci and @bhilbert4 Here ya go!

Screenshot 2024-05-07 at 11 55 46 AM

**Comment by mfixstsci** [comment]
> @mfixstsci and @bhilbert4 Here ya go!
> Screenshot 2024-05-07 at 11 55 46 AM

oh yeah, this rocks. I approve 👍

**Comment by bhilbert4** [comment]
Yeah, looks good!

**Comment by BradleySappington** [comment]
awesome, thanks!  I'll merge once its approved

File: jwql/website/apps/jwql/views.py
Comment (mfixstsci) [inline jwql/website/apps/jwql/views.py:1178]: Nice man, this is really good stuff right here.

File: jwql/website/apps/jwql/static/js/jwql.js
Comment (mfixstsci) [inline jwql/website/apps/jwql/static/js/jwql.js:333]: This loop is good too, well done.

**Review by mfixstsci — APPROVED** [review]

**Comment by mfixstsci** [comment]
> awesome, thanks! I'll merge once its approved

Send it when you are ready @BradleySappington

File: jwql/website/apps/jwql/data_containers.py
Comment (bhilbert4) [inline jwql/website/apps/jwql/data_containers.py:684]: Will there ever be a case where file_root doesn't return anything? Is the loop below enough to protect against cases like that? e.g. rootfileinfo_set will never be something like [None]?

File: jwql/website/apps/jwql/views.py
Comment (bhilbert4) [inline jwql/website/apps/jwql/views.py:68]: 👍

**Review by bhilbert4 — APPROVED** [review]
This looks good to me. I think the NIRSPec and NIRCam teams will be happy.

File: jwql/website/apps/jwql/data_containers.py
Comment (BradleySappington) [reply jwql/website/apps/jwql/data_containers.py:684]: that can theoretically happen, and if it does we will return an empty dict.

**Pull request merged**
